### ISSUE TITLE
horovod example as tensorflow_mnist.py

### ISSUE THREAD

**Issue by udacitysimon** [horovod/horovod#175]
Hi, I correctly installed horovod and can be test by run the example of tensorflow_mnist.py, first i run under np 3, which means i have 3 nodes each node has 1 GPU, total running time is 4 minutes 50 seconds till the training finished, then i run under np 1, which means i have only 1 node with 1 GPU, total runing time is almost same as np 3 case. I do not understand why the time cost is almost same, couldn't see any gain use np 3.  please advice, thanks a lot.

**Comment by glkclass** [comment]
Using more GPUs during this 'almost same running time' you should get better results (accuracy etc) comparing to single GPU.

**Comment by alsrgv** [comment]
@udacitysimon, tensorflow_mnist.py example is mostly a hello world example that demonstrates all the moving parts and is runnable on the laptop.  You can see actual speedups if you replace a model with a more powerful one.  We have an example of [ResNet50 training](https://github.com/uber/horovod/blob/master/examples/keras_imagenet_resnet50.py) which demonstrates scaling.

**Comment by udacitysimon** [comment]
@alsrgv thanks for your answer, i still have doubt that why can't see scaling in a hello world example. it should show some improvement when i use 3 GPU. 
The example of ResNet50 training is a keras example, I will try to demonstrates but my team works on tensorflow, i would like to show tensorflow_mnist.py case which exist in horovod examples.

**Comment by alsrgv** [comment]
@udacitysimon, you can add more layers to `tensorflow_mnist.py` example - then you should see 3-GPU take less wall clock time than 1-GPU.

**Comment by udacitysimon** [comment]
@alsrgv thanks for your suggestion, i will try to make test that add more layers to tensorflow_mnist.py example and report the result.  I just updated my uppper comments. Can you guide me  why each 10 step the cost is changed from 0.059 to 0.16 that use 3 GPU.

**Comment by alsrgv** [comment]
@udacitysimon, the synchronization between GPUs has non-zero cost, and if the model is too small, it may be dominating - that's most likely why you see higher step time in 3-GPU case.  In practice, you don't really need 3 GPUs to train 3-layer model - you'd either just use one GPU and train it in few mins, or train more complicated model with multiple GPUs.

**Comment by udacitysimon** [comment]
@alsrgv thanks for your answer. this make sense. So next step is add more layers to the model. will report the result. Thanks again.
